### PR TITLE
Add doc comments for FormOptions.

### PR DESCRIPTION
**Pull request by Tratcher**
@muratg @blowdart @rynowak @kichalla

### CODE REVIEW

File: src/Microsoft.AspNetCore.Http/Features/FormOptions.cs [hunk 1]
```diff
@@ -12,15 +13,65 @@ public class FormOptions
...
         public int MemoryBufferThreshold { get; set; } = DefaultMemoryBufferThreshold;
+
+        /// <summary>
+        /// If <see cref="BufferBody"/> is enabled, this is the limit for the total number of bytes that will
+        /// be buffered. Forms that exceed this limit will throw an <see cref="InvalidDataException"/> when parsed.
+        /// </summary>
         public long BufferBodyLengthLimit { get; set; } = DefaultBufferBodyLengthLimit;
+
+        /// <summary>
+        /// A limit for the number of form entries to allow. Entries with the same key will be combine.
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Http/Features/FormOptions.cs]: typo: combined

File: src/Microsoft.AspNetCore.Http/Features/FormOptions.cs [hunk 2]
```diff
@@ -12,15 +13,65 @@ public class FormOptions
...
+
+        /// <summary>
+        /// A limit for the length of the boundary identifier. Forms that exceed this limit will throw
+        /// an <see cref="InvalidDataException"/> when parsed.
+        /// </summary>
         public int MultipartBoundaryLengthLimit { get; set; } = DefaultMultipartBoundaryLengthLimit;
+
+        /// <summary>
+        /// A limit for the number of headers to allow in each multipart section. Headers with the same name will
+        /// be combine. Forms that exceed this limit will throw an <see cref="InvalidDataException"/> when parsed.
```
Comment (muratg) [inline src/Microsoft.AspNetCore.Http/Features/FormOptions.cs]: combine -> combined

**Comment by muratg** [comment]
Address the comments, and :shipit:

File: src/Microsoft.AspNetCore.Http/Features/FormOptions.cs (hunk 2 above)
Comment (kichalla) [inline src/Microsoft.AspNetCore.Http/Features/FormOptions.cs]: combined

File: src/Microsoft.AspNetCore.Http/Features/FormOptions.cs [hunk 3]
```diff
@@ -12,15 +13,65 @@ public class FormOptions
...
+        /// <summary>
+        /// If <see cref="BufferBody"/> is enabled, this many bytes of the body will be buffered in memory.
+        /// If this limit is exceeded then the buffer will be moved to a temp file on disk instead.
+        /// This also applies when buffering individual multipart section bodies.
+        /// </summary>
         public int MemoryBufferThreshold { get; set; } = DefaultMemoryBufferThreshold;
+
+        /// <summary>
+        /// If <see cref="BufferBody"/> is enabled, this is the limit for the total number of bytes that will
+        /// be buffered. Forms that exceed this limit will throw an <see cref="InvalidDataException"/> when parsed.
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Http/Features/FormOptions.cs]: curious regarding usage of `MemoryBufferThreshold` vs. `BufferBodyLengthLimit`: If I want to buffer the data and also be safe from larger payloads, I could set the MemoryBufferThreshold alone...so wondering the use of BufferBodyLengthLimit

**Comment by kichalla** [comment]
:shipit: Thanks, the comments cleared some questions for me!.

File: src/Microsoft.AspNetCore.Http/Features/FormOptions.cs [hunk 4]
```diff
@@ -12,15 +13,65 @@ public class FormOptions
...
         public int KeyCountLimit { get; set; } = FormReader.DefaultKeyCountLimit;
+
+        /// <summary>
+        /// A limit on the length of individual keys. Forms that exceed this limit will throw an
+        /// <see cref="InvalidDataException"/> when parsed.
+        /// </summary>
         public int KeyLengthLimit { get; set; } = FormReader.DefaultKeyLengthLimit;
+
+        /// <summary>
+        /// A limit on the length of individual form values. Forms that exceed this limit will throw
```
Comment (blowdart) [inline src/Microsoft.AspNetCore.Http/Features/FormOptions.cs]: If a individual form entry value exceeds this limit then whatever will throw.

File: src/Microsoft.AspNetCore.Http/Features/FormOptions.cs [hunk 5]
```diff
@@ -12,15 +13,65 @@ public class FormOptions
...
         public long BufferBodyLengthLimit { get; set; } = DefaultBufferBodyLengthLimit;
+
+        /// <summary>
+        /// A limit for the number of form entries to allow. Entries with the same key will be combine.
+        /// Forms that exceed this limit will throw an <see cref="InvalidDataException"/> when parsed.
+        /// </summary>
         public int KeyCountLimit { get; set; } = FormReader.DefaultKeyCountLimit;
+
+        /// <summary>
+        /// A limit on the length of individual keys. Forms that exceed this limit will throw an
```
Comment (blowdart) [inline src/Microsoft.AspNetCore.Http/Features/FormOptions.cs]: Forms containing a key exceeding this length will throw

File: src/Microsoft.AspNetCore.Http/Features/FormOptions.cs (hunk 3 above)
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Http/Features/FormOptions.cs]: MemoryBufferThreshold is not a hard limit, no exceptions will be thrown. It's just the amount of memory to use before switching to disk instead.

**Pull request merged**